src/io/ObjectGenerator.cpp

```diff
@@ -55,12 +56,27 @@ void ObjectGenerator::readXML(XMLfileUnits& xmlconfig) {
 
 	if(xmlconfig.changecurrentnode("velocityAssigner")) {
 		std::string velocityAssignerName;
+		bool enableRandomSeed = false;
+		int seed = 0;
 		xmlconfig.getNodeValue("@type", velocityAssignerName);
+		xmlconfig.getNodeValue("@enableRandomSeed", enableRandomSeed);
 		global_log->info() << "Velocity assigner: " << velocityAssignerName << endl;
+		if(enableRandomSeed)
+		{
+			/** A random seed for the velocity generator is created.
+			 *  The current rank is added to make sure that, if multiple simulations are instantiated across
+			 *  multiple ranks (for ex. when coupling to MaMiCo), the seeds will be unique if they are created
+			 *  at the same time
+			*/
+			seed = std::chrono::duration_cast<std::chrono::seconds>
+				(std::chrono::system_clock::now().time_since_epoch()).count() + _simulation.domainDecomposition().getRank();
+			
+		}
```

Comment:
If you want to be fancy / tidy:

```suggestion
		xmlconfig.getNodeValue("@type", velocityAssignerName);
		global_log->info() << "Velocity assigner: " << velocityAssignerName << endl;

		const int seed = [&]() {
			bool enableRandomSeed = false;
			xmlconfig.getNodeValue("@enableRandomSeed", enableRandomSeed);
			if(enableRandomSeed) {
				/** A random seed for the velocity generator is created.
				 *  The current rank is added to make sure that, if multiple simulations are instantiated across
				 *  multiple ranks (for ex. when coupling to MaMiCo), the seeds will be unique if they are created
				 *  at the same time
				 */
				return std::chrono::duration_cast<std::chrono::seconds>
				(std::chrono::system_clock::now().time_since_epoch()).count() + _simulation.domainDecomposition().getRank();
			
			} else {
				return 0;
			}
		}();
```
(check that this code is actually working. This was typed on the fly.)